docs/book/src/tasks/experimental-features/cluster-class/operate-cluster.md

```diff
@@ -266,6 +267,76 @@ Please note that:
 
 </aside>
 
+# Upgrading Cluster API
+
+There are some special considerations for ClusterClass regarding Cluster API upgrades when the upgrade includes bumping
+the apiVersion of infrastructure, bootstrap or control plane provider CRDs.
+
+In general the Cluster topology controller always uses exactly the versions of the CRDs referenced in the ClusterClass.
+This means in the following example the Cluster topology controller uses `v1beta1` even if the `DockerClusterTemplate` already 
+has a `v1beta2` version.
+
+```yaml
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: ClusterClass
+metadata:
+  name: quick-start
+  namespace: default
+spec:
+  infrastructure:
+    ref:
+      apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+      kind: DockerClusterTemplate
+...
+```
+
+It is recommended to first upgrade Cluster API and then update the apiVersions in the ClusterClass afterwards. There won't be 
+any disruptions of the reconciliation as the Cluster topology controller is able to reconcile the Cluster even with the old 
+apiVersion in the ClusterClass. 
+
+Please note that if the ClusterClass is updated before Cluster API the reconciliation will fail as the Cluster topology controller
+is unable to use resources in an apiVersion that doesn't exist yet.
```

Comment:
```suggestion
Please note that apiVersion and references in ClusterClass cannot be updated before Cluster API because the new apiVersion doesn't exist yet in the management cluster.
```